tscreen.go

```diff
@@ -681,6 +681,11 @@ func (t *tScreen) drawCell(x, y int) int {
 		if attrs&AttrStrikeThrough != 0 {
 			t.TPuts(ti.StrikeThrough)
 		}
+		if style.hyperlink != "" && ti.Hyperlink != "" {
+			t.TPuts(ti.Hyperlink + style.hyperlink + ti.StringTerminator)
+		} else if t.curstyle.hyperlink != "" && ti.Hyperlink != "" {
+			t.TPuts(ti.Hyperlink + ti.StringTerminator)
+		}
```

Comment:
I am concerned that the style code checks may cause gross inefficiencies -- we have to emit this value when changing the style, and now this string has to be part of that.

This also increases the size of the style -- which might be ok, but I've got some efficiency concerns.

I need to look at this in more depth.

Reply:
> I am concerned that the style code checks may cause gross inefficiencies -- we have to emit this value when changing the style, and now this string has to be part of that.

What do you mean? The new OSC is only emitted when the hyperlink changes -- which should happen exactly as many times as there are hyperlinks on screen. I don't think there's a better way to reduce the number of OSC sequences.

> This also increases the size of the style -- which might be ok, but I've got some efficiency concerns.

I don't think this is a big issue -- are there any benchmarks? Do you have any suggestions to improve this?

Reply:
I'm not sure the statement about when it is emitted is strictly true.

For example, what if part of your link is covered by a different style.

I also want to make sure that the check for style equality still works -- there is definitely code that tries to minimize changing attributes, and I'm not sure what the interaction will be.  Mostly I just need to dig into this to be sure this works as intended.

Reply:
So, what I really want to do is refactor this slightly to use string based capabilities in terminfo.  I think that will be a fair bit cleaner.  Other than that it looks pretty good.

Reply:
Alright, thanks for the feedback, I'll look into this.

Reply:
I'm still not loving the separate begin and end hyperlink bits... but it might be the best approach.

Reply:
There's also a problem with %s I think (possible conflation with other bits in the terminfo standard.)

Reply:
> I'm still not loving the separate begin and end hyperlink bits... but it might be the best approach.

I don't think there's a way around it. There can be arbitrary escape codes between the start of an hyperlink and its end, including e.g. color changes and other style changes. Having a single escape code for hyperlinks won't cut it.

> There's also a problem with %s I think (possible conflation with other bits in the terminfo standard.)

Any suggestions to use something else? Would `%v` be better?

Reply:
In terminfo, %s is the right thing to use here, but technically it should be used with a pop operation.  I need to review the code to see the best way to achieve what we want.